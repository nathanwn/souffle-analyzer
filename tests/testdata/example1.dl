/* A is a relation. */
.decl A(x: number, y: number)

/**
 * B is another relation.
**/
.decl B(x: number, y: number)

// Some rules for B.
B(x, y) :- A(x, y).
B(x, z) :- A(x, y), B(y, z).

.decl C(x: number, y: number)
C(x, y) :- A(x, y), !B(x, y).

.input A
.output C
