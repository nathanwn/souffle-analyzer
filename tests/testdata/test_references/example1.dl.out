example1.dl
 0 | /* A is a relation. */
 1 | .decl A(x: number, y: number)
 2 |
 3 | /**
 4 |  * B is another relation.
 5 | **/
 6 | .decl B(x: number, y: number)
 7 |
 8 | // Some rules for B.
 9 | B(x, y) :- A(x, y).
10 | B(x, z) :- A(x, y), B(y, z).
11 |
12 | .decl C(x: number, y: number)
13 | C(x, y) :- A(x, y), !B(x, y).
14 |
15 | .input A
16 | .output C

=====

-- Cursor range --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~
-- References --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~

example1.dl
15 | .input A
   |        ~

_____

-- Cursor range --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~
-- References --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~

_____

-- Cursor range --
example1.dl
 9 | B(x, y) :- A(x, y).
   | ~
-- References --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~

_____

-- Cursor range --
example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~
-- References --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~

example1.dl
15 | .input A
   |        ~

_____

-- Cursor range --
example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~
-- References --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~

_____

-- Cursor range --
example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~
-- References --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~

example1.dl
15 | .input A
   |        ~

_____

-- Cursor range --
example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~
-- References --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~

_____

-- Cursor range --
example1.dl
12 | .decl C(x: number, y: number)
   |       ~
-- References --
example1.dl
12 | .decl C(x: number, y: number)
   |       ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   | ~

example1.dl
16 | .output C
   |         ~

_____

-- Cursor range --
example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   | ~
-- References --
example1.dl
12 | .decl C(x: number, y: number)
   |       ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   | ~

example1.dl
16 | .output C
   |         ~

_____

-- Cursor range --
example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~
-- References --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~

example1.dl
15 | .input A
   |        ~

_____

-- Cursor range --
example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~
-- References --
example1.dl
 6 | .decl B(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   | ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |                     ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |                      ~

_____

-- Cursor range --
example1.dl
15 | .input A
   |        ~
-- References --
example1.dl
 1 | .decl A(x: number, y: number)
   |       ~

example1.dl
 9 | B(x, y) :- A(x, y).
   |            ~

example1.dl
10 | B(x, z) :- A(x, y), B(y, z).
   |            ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   |            ~

example1.dl
15 | .input A
   |        ~

_____

-- Cursor range --
example1.dl
16 | .output C
   |         ~
-- References --
example1.dl
12 | .decl C(x: number, y: number)
   |       ~

example1.dl
13 | C(x, y) :- A(x, y), !B(x, y).
   | ~

example1.dl
16 | .output C
   |         ~

_____
